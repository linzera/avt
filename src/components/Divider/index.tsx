import React from 'react';
import {View, StyleSheet, StyleProp, ViewStyle} from 'react-native';
import colors from '~/theme/colors';

const styles = StyleSheet.create({
  divider: {height: 2, backgroundColor: colors.gray200},
  horizontal: {
    width: 2,
    backgroundColor: colors.gray200,
  },
});

interface Props {
  style?: StyleProp<ViewStyle>;
  horizontal?: boolean;
}

export default function Divider(props: Props) {
  return (
    <View
      style={[
        props.horizontal ? styles.horizontal : styles.divider,
        props.style,
      ]}
    />
  );
}
