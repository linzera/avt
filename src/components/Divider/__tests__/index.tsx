/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import {render} from '@testing-library/react-native';
import Divider from '..';

describe('<Divider />', () => {
  it('should render properly given default props', () => {
    const {toJSON} = render(<Divider />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render properly given horizontal prop', () => {
    const {toJSON} = render(<Divider horizontal />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render properly given style prop', () => {
    const {container} = render(<Divider style={{height: 50}} />);
    expect(container).toHaveStyle({height: 50});
  });
});
