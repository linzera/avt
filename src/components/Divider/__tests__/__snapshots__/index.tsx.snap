// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> should render properly given default props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#eeeeee",
        "height": 2,
      },
      undefined,
    ]
  }
/>
`;

exports[`<Divider /> should render properly given horizontal prop 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#eeeeee",
        "width": 2,
      },
      undefined,
    ]
  }
/>
`;
